ee/tabby-webserver/src/service/answer.rs

```diff
@@ -212,23 +224,108 @@ impl AnswerService {
         );
 
         let mut params = params.clone();
-        override_params
-            .as_ref()
-            .inspect(|x| x.override_params(&mut params));
+        if let Some(override_params) = override_params {
+            override_params.override_params(&mut params);
+        }
 
         match self.code.search_in_language(query, params).await {
-            Ok(docs) => docs.hits,
-            Err(err) => {
-                if let CodeSearchError::NotReady = err {
-                    debug!("Code search is not ready yet");
-                } else {
-                    warn!("Failed to search code: {:?}", err);
-                }
+            Ok(docs) => {
+                self.merge_code_snippets(docs.hits, &policy, source_id)
+                    .await
+            }
+            Err(e) => {
+                warn!("Error in code search: {:?}", e);
                 vec![]
             }
         }
     }
 
+    /// Merges code snippets from search hits, potentially combining multiple hits from the same file.
+    /// If multiple hits are found in the same file, the function will attempt to merge them into a single hit.
+    async fn merge_code_snippets(
```

Comment:
i recommend extract this as an individual function (instead of a method) for ease of testing.